engines/dotnet/Main.cs

```diff
@@ -159,70 +140,74 @@ static void Main(string[] args)
         //
         // So we do this dance with raw bytes, which is just amazingly
         // inconvenient, because C# doesn't have byte string support.
-        Stream stdin = Console.OpenStandardInput();
-        List<byte> raw = new List<byte>();
-        byte[] buf = new byte[8192];
-        int nread = 0;
-        while ((nread = stdin.Read(buf, 0, buf.Length)) > 0) {
-            for (int i = 0; i < nread; i++) {
-                raw.Add(buf[i]);
-            }
+        MemoryStream stdinCopy = new();
+        using (Stream stdin = Console.OpenStandardInput()) {
+            stdin.CopyTo(stdinCopy);
         }
+        ReadOnlySpan<byte> raw = stdinCopy.GetBuffer().AsSpan(0, (int)stdinCopy.Length);
+
         // OK, now read each of our KLV items and build up our config.
-        Config config = new Config(args[0]);
-        while (raw.Count > 0) {
+        Config config = new(args[0]);
+        while (!raw.IsEmpty) {
             var klv = new OneKLV(raw);
-            raw = raw.GetRange(klv.len, raw.Count - klv.len);
-            if (klv.key == "name") {
-                config.name = klv.value;
-            } else if (klv.key == "model") {
-                config.model = klv.value;
-            } else if (klv.key == "pattern") {
-                if (config.pattern != null) {
-                    throw new Exception("only one pattern is supported");
-                }
-                config.pattern = klv.value;
-            } else if (klv.key == "case-insensitive") {
-                config.caseInsensitive = klv.value == "true";
-            } else if (klv.key == "unicode") {
-                config.unicode = klv.value == "unicode";
-            } else if (klv.key == "haystack") {
-                config.haystack = klv.value;
-            } else if (klv.key == "max-iters") {
-                config.maxIters = int.Parse(klv.value);
-            } else if (klv.key == "max-warmup-iters") {
-                config.maxWarmupIters = int.Parse(klv.value);
-            } else if (klv.key == "max-time") {
-                config.maxTime = long.Parse(klv.value);
-            } else if (klv.key == "max-warmup-time") {
-                config.maxWarmupTime = long.Parse(klv.value);
-            } else {
-                throw new Exception($"unrecognized KLV key {klv.key}");
+            raw = raw.Slice(klv.len);
+            switch (klv.key)
```

Comment:
pattern matching?

Reply:
You mean like:
```C#
something = klv.key switch { ... }
```
? That works when the "something" is the same for all branches, but it's not here.

Reply:
Ah - thanks